### PR TITLE
Travis support

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

**Comment by Mic92** [comment]
one test requires `nixpkgs` in its path otherwise the following would work:

```diff
diff --git a/default.nix b/default.nix
index 6884cb6..424d9a1 100644
--- a/default.nix
+++ b/default.nix
@@ -15,7 +15,7 @@ let
       sha256 = "0xa340bz8r3s88k91dldags32f3js8gqmcjsrsrn0p514kiacjsn";
     };
   };
-in buildGoPackage {
+in buildGoPackage rec {
   name = "go-nix";
   goPackagePath = "github.com/orivej/go-nix";
   src = ./.;
@@ -25,4 +25,8 @@ in buildGoPackage {
     ragel
     goyacc
   ];
+  doCheck = true;
+  checkPhase = ''
+    (cd $NIX_BUILD_TOP/go/src/${goPackagePath} && go test ./...)
+  '';
 }
```

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Overview
 
+[![Build Status](https://travis-ci.com/orivej/go-nix.svg?branch=master)](https://travis-ci.com/orivej/go-nix)
```
Comment (Mic92) [inline README.md:3]: You also need to enable your project in travis.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+language: nix
+script:
+  - nix-shell --command "GOCACHE= go test ./..."
```
Comment (orivej) [inline .travis.yml]: Why did you add `GOCACHE=`? This may break with Go 1.12: https://golang.org/doc/go1.11#gocache

This does not work with `nix-shell --pure`, so it probably won't work in Travis.

**Comment by orivej** [comment]
> one test requires `nixpkgs` in its path otherwise the following would work

Fixed in 63f0d7c81d18149b474b8ad78dce6ffd2f90ffcf

**Comment by Mic92** [comment]
Also it is fixed, you probably still want to run all tests on travis I suppose.

File: .travis.yml (hunk 2 above)
Comment (Mic92) [reply .travis.yml]: Because the nix-shell set `GOCACHE=off`:

```console
$ nix-shell --command "echo $GOCACHE" 
off
```

This would break the build in this case.

Comment (Mic92) [reply .travis.yml]: `nix-shell --pure` does not allow network access, travis does.

Comment (Mic92) [reply .travis.yml]: It now sets `GOCACHE=` to an tempdir instead. Also `GOCACHE=` should still work with go 1.12 because it does not disable the cache dir and rather use an default value.

**Comment by orivej** [comment]
Thank you!

The initial build still failed, like in `nix-shell --pure`: https://travis-ci.org/orivej/go-nix/builds/422390619

> `nix-shell --pure` does not allow network access, travis does.

AFAIK it only cleans environment variables.

> you probably still want to run all tests on travis I suppose.

Yes, I have done this in https://github.com/orivej/go-nix/commit/ab083a5d8416c743f9b4f2a4c310e5e92334df7f
